src/soundvectors.py

```diff
@@ -589,32 +797,19 @@ def get_vec(self, sound, vectorize=True):
         base_vec = self[sound]
         return base_vec.as_vector() if vectorize else base_vec.as_dict()
 
-    def _get_hierarchy_level(self, domain):
-        if domain in self.feature_hierarchy:
-            return self.feature_hierarchy.index(domain)
-        return len(self.feature_hierarchy)
-
-    def _get_features(self, featureset):
-        primary_features, secondary_features = [], []
-        for feature in featureset:
-            if self.feature_values.get(feature, {"domain": ""})["domain"] in self.feature_hierarchy:
-                primary_features.append(feature)
-            else:
-                secondary_features.append(feature)
-        primary_features = sorted(
-            primary_features,
-            key=lambda x: self._get_hierarchy_level(
-                self.feature_values.get(x, {"domain": ""})["domain"]))
-
-        return primary_features, secondary_features
-
-    def _apply_feature(self, feature, base_vec, valid=(1, -1)):
-        bin_feature_vec = self.feature_values.get(
-            feature, {"features": FeatureBundle()})["features"]
-        return base_vec.replace(**bin_feature_vec.as_dict(valid_values=valid))
+    def _get_features(self, featureset) -> typing.Tuple[typing.List, typing.List]:
+        primary, secondary = [], []
+        # Augment the featureset with matching HierarchicalFeatures if they are defined:
+        featureset = [
+            (feature, self.feature_values.get(feature))
+            for feature in featureset if feature in self.feature_values]
```

Comment:
Here, I just filter out all features that cannot be mapped to a `FeatureBundle`.